ext/pbr_ext/encode.h

```diff
@@ -0,0 +1,105 @@
+#ifndef encode_h__
+#define encode_h__
+
+#include <cstdint>
+#include "pbr_ext.h"
+
+inline void w_varint32_bytes(buf_t& buf, uint32_t offset, uint32_t n, int32_t min_bytes) {
+  uint32_t i = 0;
+  min_bytes--;
+  while (n > 127 || min_bytes > 0) {
+    buf[offset + i] = (uint32_t)((n & 127) | 128);
+    n >>= 7;
```

Comment:
iirc, the varint and zigzag functions are the one part of the c++ code that I stole from examples online :)